src/components/TaskDetail.jsx

```diff
@@ -361,6 +443,148 @@ export function TaskDetail({ task, isOpen, onClose, onUpdate, onDelete, onAddLab
+                  />
+                </div>
+              ) : (
+                <button
+                  className="add-subtask-button"
+                  onClick={() => setIsAddingSubtask(true)}
+                >
+                  + Add subtask
+                </button>
+              )}
+            </div>
+
+            <div className="dependencies-section">
+              <div className="section-title">Dependencies</div>
+              <div className="dependency-list">
+                {(task.dependencies || []).map(dependency => (
+                  <div key={dependency.id} className="dependency-item">
+                    <span className={`dependency-type ${DEPENDENCY_TYPES[dependency.type].class}`}>
+                      {DEPENDENCY_TYPES[dependency.type].label}
+                    </span>
+                    <div className="dependency-content">
+                      <div className={`dependency-status ${getDependencyStatus(dependency)}`} />
+                      <span>{dependency.content}</span>
+                    </div>
+                    {user && (
+                      <button
+                        className="delete-subtask"
+                        onClick={() => handleRemoveDependency(dependency.id)}
+                        title="Remove dependency"
+                      >
+                        ×
+                      </button>
+                    )}
+                  </div>
+                ))}
+              </div>
+
+              {user && !showDependencySearch ? (
+                <button
+                  className="add-dependency-button"
+                  onClick={() => setShowDependencySearch(true)}
+                >
+                  + Add dependency
+                </button>
+              ) : user && (
+                <div className="dependency-search">
+                  <select
+                    className="dependency-type-select"
+                    value={selectedDependencyType}
+                    onChange={(e) => setSelectedDependencyType(e.target.value)}
+                  >
+                    {Object.entries(DEPENDENCY_TYPES).map(([key, { label }]) => (
+                      <option key={key} value={key}>{label}</option>
+                    ))}
+                  </select>
+                  <input
+                    type="text"
+                    className="dependency-search-input"
+                    value={dependencySearch}
+                    onChange={(e) => setDependencySearch(e.target.value)}
+                    placeholder="Search for tasks..."
+                    autoFocus
+                  />
+                  <div className="dependency-search-results">
+                    {Object.values(columns).map(column => 
+                      column.tasks
+                        ?.filter(t => 
+                          t.id !== task.id && 
+                          t.content.toLowerCase().includes(dependencySearch.toLowerCase())
+                        )
```

Comment:
![kody code-review](https://img.shields.io/badge/kody-code--review-brightgreen)

```javascript
column.tasks?.filter(t => t.id !== task.id && t.content.toLowerCase().includes(debouncedSearch.toLowerCase())) // Add debouncing and memoization
```
The dependency search filter performs case-insensitive string matching on every keystroke. Consider debouncing the search input and memoizing the filtered results to improve performance with large datasets.



<sub>Talk to Kody by mentioning @kody</sub>

<sub>Was this suggestion helpful? React with 👍 or 👎 to help Kody learn from this interaction.</sub><!-- kody-codereview -->&#8203;
&#8203;